soc/arm/st_stm32/stm32l0/power.c

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2021 Fabio Baltieri
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr.h>
+#include <pm/pm.h>
+#include <soc.h>
+#include <init.h>
+
+#include <stm32l0xx_ll_utils.h>
+#include <stm32l0xx_ll_bus.h>
+#include <stm32l0xx_ll_cortex.h>
+#include <stm32l0xx_ll_pwr.h>
+#include <stm32l0xx_ll_rcc.h>
+#include <stm32l0xx_ll_system.h>
+#include <clock_control/clock_stm32_ll_common.h>
+
+#include <logging/log.h>
+LOG_MODULE_DECLARE(soc, CONFIG_SOC_LOG_LEVEL);
+
+/* Select MSI as wake-up system clock if configured, HSI otherwise */
+#if STM32_SYSCLK_SRC_MSI
+#define RCC_STOP_WAKEUPCLOCK_SELECTED LL_RCC_STOP_WAKEUPCLOCK_MSI
+#else
+#define RCC_STOP_WAKEUPCLOCK_SELECTED LL_RCC_STOP_WAKEUPCLOCK_HSI
+#endif
+
+/* Invoke Low Power/System Off specific Tasks */
+void pm_power_state_set(struct pm_state_info info)
+{
+	switch (info.state) {
+	case PM_STATE_SUSPEND_TO_IDLE:
```

Comment:
I think you need `LL_PWR_ClearFlag_WU` not only for STANDBY but for reliably entering STOP mode as well.

Reply:
Hey I rigged up a test for that and it seems to me like it's not necessary, i.e. if I don't clear the flag before a standby, the CPU instantly wakes up, but for STOP it seems like the flag gets set but the CPU does not wake up on it. I can set it if you think I'm missing something, but then if the application code wants to read the flag value, it has to be done in the early system initialization, before any PM sleep kicks in. Let me know what you think.

Reply:
In my test it also seems to work without it, but all documentation I have found on entering STOP mode on L0-series mentions clearing WUF as a requirement, so I'm hesitant to leave it out.

Reply:
Fair enough, let's add it.